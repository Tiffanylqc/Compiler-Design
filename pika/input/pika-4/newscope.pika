static var a := 5.
static var b := a + 7.

func f<int x> -> void
{
	static var x := x.
	print x;.
	return.
}

exec
{
	while (_true_)
	{
		static var test := 2.
		if (test > 0) { print 999;. }
		test := test - 1.
		if (test < 0) { break. }
	}
	call f(2).
	call f(3).
	call f(00).
	print a;b;_n_.
}
