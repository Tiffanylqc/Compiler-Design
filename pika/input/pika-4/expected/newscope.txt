999 999 2 2 2 5 12 
